// xendit-node/invoice/models/DirectDebitType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DirectDebitTypeToJSON = exports.DirectDebitTypeFromJSONTyped = exports.DirectDebitTypeFromJSON = exports.DirectDebitType = void 0;
/**
 * Representing the available Direct Debit channels used for invoice-related transactions.
 * @export
 */
exports.DirectDebitType = {
    BaBri: 'BA_BRI',
    DcBri: 'DC_BRI',
    DdBri: 'DD_BRI',
    DdMandiri: 'DD_MANDIRI',
    BaBpi: 'BA_BPI',
    DcBpi: 'DC_BPI',
    DdBpi: 'DD_BPI',
    BaUbp: 'BA_UBP',
    DcUbp: 'DC_UBP',
    DdUbp: 'DD_UBP',
    BcaKlikpay: 'BCA_KLIKPAY',
    BaBcaKlikpay: 'BA_BCA_KLIKPAY',
    DcBcaKlikpay: 'DC_BCA_KLIKPAY',
    DdBcaKlikpay: 'DD_BCA_KLIKPAY',
    DdBdoEpay: 'DD_BDO_EPAY',
    DdRcbc: 'DD_RCBC',
    DdChinabank: 'DD_CHINABANK',
    BaChinabank: 'BA_CHINABANK',
    DcChinabank: 'DC_CHINABANK',
    XenditEnumDefaultFallback: "UNKNOWN_ENUM_VALUE"
};
function DirectDebitTypeFromJSON(json) {
    return DirectDebitTypeFromJSONTyped(json, false);
}
exports.DirectDebitTypeFromJSON = DirectDebitTypeFromJSON;
function DirectDebitTypeFromJSONTyped(json, ignoreDiscriminator) {
    if (json !== "" || json !== null) {
        const key = Object.keys(exports.DirectDebitType)[Object.values(exports.DirectDebitType).indexOf(json)];
        return exports.DirectDebitType[key] === undefined ?
            exports.DirectDebitType['XenditEnumDefaultFallback'] : exports.DirectDebitType[key];
    }
    return json;
}
exports.DirectDebitTypeFromJSONTyped = DirectDebitTypeFromJSONTyped;
function DirectDebitTypeToJSON(value) {
    return value;
}
exports.DirectDebitTypeToJSON = DirectDebitTypeToJSON;
